lib/galaxy/tools/parameters/basic.py

```diff
@@ -1853,12 +1864,17 @@ def append(list, hda, name, src, keep=False):
         # add dataset collections
         dataset_collection_matcher = dataset_matcher_factory.dataset_collection_matcher(dataset_matcher)
         for hdca in history.active_visible_dataset_collections:
-            match = dataset_collection_matcher.hdca_match(hdca, reduction=multiple)
+            match = dataset_collection_matcher.hdca_match(hdca)
             if match:
+                subcollection_type = None
+                if multiple:
+                    subcollection_type = self._history_query(trans).can_map_over(hdca).collection_type
```

Comment:
Something isn't quite right here, I see this traceback when building the tool form:

```
galaxy.web.framework.decorators ERROR 2018-06-01 10:45:31,618 Uncaught exception in exposed API method:
Traceback (most recent call last):
  File "/Users/mvandenb/src/galaxy/lib/galaxy/web/framework/decorators.py", line 281, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "/Users/mvandenb/src/galaxy/lib/galaxy/webapps/galaxy/api/tools.py", line 113, in build
    return tool.to_json(trans, kwd.get('inputs', kwd))
  File "/Users/mvandenb/src/galaxy/lib/galaxy/tools/__init__.py", line 1883, in to_json
    self.populate_model(request_context, self.inputs, state_inputs, tool_model['inputs'])
  File "/Users/mvandenb/src/galaxy/lib/galaxy/tools/__init__.py", line 1954, in populate_model
    tool_dict = input.to_dict(request_context)
  File "/Users/mvandenb/src/galaxy/lib/galaxy/tools/parameters/basic.py", line 1871, in to_dict
    subcollection_type = self._history_query(trans).can_map_over(hdca).collection_type
AttributeError: 'bool' object has no attribute 'collection_type'
```

Reply:
Yes - the other place this is used in that file it verifies it is truthy before using it - my bad. I'll throw in a continue in that loop if it is false.

Reply:
I see you've rebased the commit, but something must've gone wrong again.
I suppose this should be as in 880c0217ab3d007e34431c6ae54101f89c1aa3b4 ?